### ISSUE TITLE
Add some useful predicates

### ISSUE THREAD

**Issue by chessai** [chessai/semirings#37]
I think the following are useful. I have had uses for them in the past:

```
isZero :: Semiring a => a -> Bool
isZero x = x == zero
isOne :: Semiring a => a -> Bool
isOne x = x == one
isUnit :: Semiring a => a -> Bool
isUnit x = isZero x || isOne x
```

**Comment by Bodigrim** [comment]
I am not sure about `isUnit`: what is the use case for it? Also its names is confusing: one may rightfully expect that this is a predicate for [units in a ring](https://en.wikipedia.org/wiki/Unit_(ring_theory)).

**Comment by chessai** [comment]
Oh, right - sorry, I was tired. I meant for isUnit to test for unitness of a field (as in the link),  but we cannot actually have this function do what we want given the formulation of Field proposed in #36

**Comment by chessai** [comment]
resolved in 665d982e218c7851fd27a7cec32edd9ab98d6b0c
